Removed thread_local from RandomNumberGeneratorService

Dr15Jones opened the pull request:
The thread_local in RandomNumberGeneratorService was being used to
support a backwards compatible interface. The old interface is also
supported for certain transitions via another thread_local from
the framework. It appears that the old interface is no longer being
called by non-framework supported transitions.

Dr15Jones committed:
Removed thread_local from RandomNumberGeneratorService

The thread_local in RandomNumberGeneratorService was being used to
support a backwards compatible interface. The old interface is also
supported for certain transitions via another thread_local from
the framework. It appears that the old interface is no longer being
called by non-framework supported transitions.

cmsbuild commented:
A new Pull Request was created by @Dr15Jones (Chris Jones) for CMSSW_8_0_X.

It involves the following packages:

IOMC/RandomEngine


@cmsbuild, @smuzaffar, @Dr15Jones, @davidlange6 can you please review it and eventually sign? Thanks.
@wddgit, @wmtan this is something you requested to watch as well.

Following commands in first line of a comment are recognized

* **+1|approve[d]|sign[ed]**: L1/L2's to approve it
* **-1|reject[ed]**: L1/L2's to reject it
* **assign &lt;category&gt;[,&lt;category&gt;[,...]]**: L1/L2's to request signatures from other categories
* **unassign &lt;category&gt;[,&lt;category&gt;[,...]]**: L1/L2's to remove signatures from other categories
* **hold**: L1/L2's to mark it as on hold
* **merge**: L1 to merge this request
* **[@cmsbuild,] please test**: L1/L2 and selected users to start jenkins tests
* **[@cmsbuild,] please test with cms-sw/cmsdist#&lt;PR&gt;**: L1/L2 and selected users to start jenkins tests using externals from cmsdist

@degano you are the release manager for this.
You can merge this pull request by typing 'merge' in the first line of your comment.

Dr15Jones commented:
please test

Dr15Jones commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_0_X IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

cmsbuild commented:
-1 
 
Tested at: 67ef2a9b26437601e9d7870cc8887b8fcad3a06d
I found errors in the following unit tests: 
 
 ---> test testDataFormatsPatCandidates had ERRORS
---> test TestIOMCRandomEngineService had ERRORS
 
 
you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12384/9657/summary.html 

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/4d90d57068dfd7ef2bf5b9e7be7ce29e61bf6b0c
https://github.com/cms-sw/cmssw/commit/d85491c42af33996e9c8550328e3c9c67a45934e
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12384/9657/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12384/9657/git-merge-result

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_0_X IBs (but tests are reportedly failing). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

Dr15Jones committed:
Updated test to match new behavior

We no longer support calling RandomNumberGenerate::mySeed at
specific framework transitions. Test now enforces the change.

Dr15Jones commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #12384 was updated. @cmsbuild, @smuzaffar, @Dr15Jones, @davidlange6 can you please check and sign again.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12384/9657/summary.html

cmsbuild commented:
-1 
 
Tested at: aaa76e475348c9b2af84fe0ca593891e67601c03
I found errors in the following unit tests: 
 
 ---> test testDataFormatsPatCandidates had ERRORS
 
 
you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12384/9665/summary.html 

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/4d90d57068dfd7ef2bf5b9e7be7ce29e61bf6b0c
https://github.com/cms-sw/cmssw/commit/d85491c42af33996e9c8550328e3c9c67a45934e
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12384/9665/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12384/9665/git-merge-result

Dr15Jones commented:
The test failure is not from this pull request

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12384/9665/summary.html

Dr15Jones commented:
please test

Dr15Jones commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_0_X IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

cmsbuild commented:
+1
Tested at: aaa76e475348c9b2af84fe0ca593891e67601c03
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12384/9777/summary.html

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_0_X IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12384/9777/summary.html

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_0_X IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

davidlange6 commented:
+1
